MDANSE/Src/MDANSE/Framework/AtomSelector/selector.py

```diff
@@ -13,362 +13,234 @@
-        self.match_exists = self.create_default_settings()
-        for k0, v0 in self.match_exists.items():
-            if isinstance(v0, dict):
-                for k1 in v0.keys():
-                    self.match_exists[k0][k1] = True
-            else:
-                self.match_exists[k0] = self._funcs[k0](
-                    self.trajectory, check_exists=True
-                )
 
-        self.settings = self.create_default_settings()
-
-    def create_default_settings(self) -> dict[str, Union[bool, dict]]:
-        """Create a new settings dictionary with default settings.
-
-        Returns
-        -------
-        dict[str, Union[bool, dict]]
-            A settings dictionary.
         """
-        settings = copy.deepcopy(self._default)
-        for k, vs in self._kwarg_vals.items():
-            for v in sorted(vs):
-                settings[k][v] = False
-        return settings
-
-    def reset_settings(self) -> None:
-        """Resets the settings back to the defaults."""
-        self.settings = self.create_default_settings()
-
-    def update_settings(
-        self, settings: dict[str, Union[bool, dict]], reset_first: bool = False
-    ) -> None:
-        """Updates the selection settings.
+        self.reset()
+
+    def reset(self):
+        """Initialise the attributes to an empty list of operations."""
+        self.system = None
+        self.trajectory = None
+        self.all_idxs = set()
+        self.operations = {}
+
+    def set_selection(
+        self,
+        *,
+        number: Optional[int] = None,
+        function_parameters: dict[str, Any],
+    ):
+        """Append a new selection operation, or overwrite an existing one.
 
         Parameters
         ----------
-        settings : dict[str, bool | dict]
-            The selection settings.
-        reset_first : bool, optional
-            Resets the settings to the default before loading.
-
-        Raises
-        ------
-        ValueError
-            Raises a ValueError if the inputted settings are not valid.
-        """
-        if not self.check_valid_setting(settings):
-            raise ValueError(
-                f"Settings are not valid for the given chemical system - {settings}."
-            )
+        number : Union[int, None], optional
+            the position of the new selection in the sequence of operations
+        function_parameters : Dict[str, Any], optional
+            the dictionary of keyword arguments defining a selection operation
 
-        if reset_first:
-            self.reset_settings()
-
-        for k0, v0 in settings.items():
-            if isinstance(self.settings[k0], dict):
-                for k1, v1 in v0.items():
-                    self.settings[k0][k1] = v1
-            else:
-                self.settings[k0] = v0
+        """
+        if number is None:
+            number = len(self.operations)
+        else:
+            number = int(number)
```

Comment:
If it's not too long an option is:
```suggestion
        number = int(number) if number is not None else len(self.operations)
```